src/value.rs

```diff
@@ -183,6 +184,24 @@ impl<'s> Value<'s> {
         }
         .map_err(DataError::EncodingError)
     }
+
+    pub fn serialized_size(&self) -> Result<u64, DataError> {
+        match self {
+            Value::Bool(v) => serialized_size(&(Type::Bool.to_tag(), *v)),
+            Value::U64(v) => serialized_size(&(Type::U64.to_tag(), *v)),
+            Value::I64(v) => serialized_size(&(Type::I64.to_tag(), *v)),
+            Value::F64(v) => serialized_size(&(Type::F64.to_tag(), v.0)),
+            Value::Instant(v) => serialized_size(&(Type::Instant.to_tag(), *v)),
+            Value::Str(v) => serialized_size(&(Type::Str.to_tag(), v)),
+            Value::Json(v) => serialized_size(&(Type::Json.to_tag(), v)),
+            Value::Blob(v) => serialized_size(&(Type::Blob.to_tag(), v)),
+            Value::Uuid(v) => {
+                // Processed above to avoid verbose duplication of error transforms.
```

Comment:
Erm, one nit: this comment, which is also in `to_bytes()`, appears to be incorrect for both `to_bytes()` and `serialized_size()`, neither of which actually processes `Type::Uuid` earlier in the function.

It looks like the comment comes from `from_type_and_data()`, which does process `Type::Uuid` separately (i.e. before the general match statement).

Reply:
Copy&pasta oops! I'll fix it up.